File: 3258-count-substrings-that-satisfy-k-constraint-i/3258-count-substrings-that-satisfy-k-constraint-i.java
class Solution {
    public int countKConstraintSubstrings(String s, int k) {
        int result = 0;
        for (int i = 0; i < s.length(); i++) {
            int[] count = new int[2];
            for (int j = i; j < s.length(); j++) {
                count[s.charAt(j) - '0']++;
                result += count[0] > k && count[1] > k ? 0 : 1;
            }
}
        return result;
    }
}

File: 3258-count-substrings-that-satisfy-k-constraint-i/3258-count-substrings-that-satisfy-k-constraint-i.py
class Solution:
    def countKConstraintSubstrings(self, s: str, k: int) -> int:
        n, ans = len(s), 0
        
        for i in range(n):
            a, b = 0, 0
            for j in range(i, n):
                if s[j] == '1':
                    b = b+1
                else:
                    a = a+1
                
                if a <= k or b <= k:
                    ans += 1
                else:
                    break
        
        return ans